### DISCUSSION TITLE
run an unikernel in a container

### DISCUSSION THREAD

**Discussion by qsmen** [kata-containers/kata-containers#8736] [Ideas]
Kata container is securer partly because  an independent kernel is run in a container. How about to run a unikernel in a container?
It seems possible. In stead of loading an independent os kernel, now load a unikernel into the container.  As a unikernel is selfcontained it is executable. As for  io syscall, they can be  passed to hypervisor.
If feasible,   that would be faster, securer, smaller.  
By the way, I did not do a search because it depends on luck to login github.
Thank you.

**Comment by qdrddr** [comment]
Added a feature request
https://github.com/kata-containers/kata-containers/issues/13817
